### PR TITLE
Fix issue#357

### PR DESCRIPTION
**Pull request by Psycho7**
Issues #357 Will allocate 4 GB memory if actual physical memory size is not available on windows

### CODE REVIEW

File: taichi/unified_allocator.cpp [hunk 1]
```diff
@@ -83,10 +83,16 @@ void taichi::Tlang::UnifiedAllocator::create(bool gpu) {
...
-  phys_mem_size /= 1024;                               // MB
-  TC_INFO("Physical memory size {} MB", phys_mem_size);
+  std::size_t phys_mem_size;  
+  if (GetPhysicallyInstalledSystemMemory(&phys_mem_size)) {   // KB
+    phys_mem_size /= 1024;                                    // MB
+    TC_INFO("Physical memory size {} MB", phys_mem_size);
+  } else {
+    auto err = GetLastError();
+    /* Error Codes: https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes */
+    TC_ERROR("Unable to get physical memory size [ Win32 Error Code {} ].", err);
```
Comment (yuanming-hu) [inline taichi/unified_allocator.cpp]: `TC_ERROR` will halt the program. `TC_WARN` seems a better fit here.

**Comment by yuanming-hu** [comment]
Thanks for the PR! It generally looks good to me. Are you able to reproduce the original error on your end and confirm it is fixed now? If not we can release v0.3.22 and ask the Windows users to test it.

File: taichi/unified_allocator.cpp (hunk 1 above)
Comment (Psycho7) [reply taichi/unified_allocator.cpp]: My bad. Thought it was just log level.

**Comment by Psycho7** [comment]
Not able to reproduce. Seems like malformed BIOS data is the reason, not sure how to produce it...

**Comment by yuanming-hu** [comment]
Yeah, don't destroy your BIOS just to reproduce it :-) I'll release v0.3.22 and ask people to test it then. Thank you again for contributing!

**Pull request merged**